h/static/scripts/controllers/form-input-controller.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var { setElementState } = require('../util/dom');
+
+/**
+ * Controller for form input fields.
+ */
+class FormInputController extends Controller {
+  constructor(element, options) {
+    super(element, options);
+
+    this.refs.formInput.addEventListener('focus', () => {
+      this.trigger('form-input:focus');
```

Comment:
I stand by my earlier assessment that the division of labour here between `FormInputController` and `FormController` is very awkward.

The only thing this controller does is mirror DOM events to our own homebrew event system, and mirror the FormController's `editing` state boolean onto this controller. I just don't see how that's making anything any easier, or any clearer.